.github/workflows/integration.yml

```diff
@@ -20,8 +20,8 @@ jobs:
     steps:
       - uses: actions/checkout@v3
 
-      - name: Run sql docker-compose.integration
-        run: docker-compose --file=test/OpenTelemetry.Instrumentation.SqlClient.Tests/docker-compose.yml --file=build/docker-compose.${{ matrix.version }}.yml --project-directory=. up --exit-code-from=tests --build
+      - name: Run sql integration
+        run: dotnet test test/OpenTelemetry.Instrumentation.SqlClient.Tests/OpenTelemetry.Instrumentation.SqlClient.Tests.csproj --framework ${{ matrix.version }} --filter CategoryName=SqlIntegrationTests
```

Comment:
We can probably get rid of the separate workflow step now as these tests should just run with the rest of the unit tests.

Reply:
But you still need the filter. I have tried running all tests without the filter, but the tests never complete. It seems like the remaining tests are waiting for the Docker Compose environment to be available. Please note that the filter already exists and is being used [here](https://github.com/open-telemetry/opentelemetry-dotnet/blob/d924663dc3d6bfdf737bc49ceaa1addcec90a2d6/test/OpenTelemetry.Instrumentation.SqlClient.Tests/docker-compose.yml#L20), and was not introduced by me. I believe that there is another stage where the remaining tests run (see the following [Dockerfile](https://github.com/open-telemetry/opentelemetry-dotnet/blob/d924663dc3d6bfdf737bc49ceaa1addcec90a2d6/test/OpenTelemetry.Instrumentation.SqlClient.Tests/Dockerfile)). Thanks to Testcontainers, all these configurations are not necessary anymore.

Please note that the initial goal was to enable the disabled SQL tests and not to refactor all tests, etc. in the first place.

Reply:
Oh, my bad. I believe you are correct in that the tests are already run with `build-test`. However, they fail on Windows because the Docker image is not available (compatible) for the Windows engine.

Reply:
> However, they fail on Windows because the Docker image is not available (compatible) for the Windows engine.

I still haven't had a chance to look too closely at TestContainers for .NET. Is it that they just aren't compatible with Windows at all? Or is it specifically the SQL image that is not compatible?

Reply:
The original integration tests only ran on linux anyways. So maybe what we could do is introduce a new xUnit attribute that will skip the test if not linux. Something similar to the `SkipUnlessEnvVarFoundTheory` but instead `SkipIfNotLinux`.

Reply:
> I still haven't had a chance to look too closely at TestContainers for .NET. Is it that they just aren't compatible with Windows at all? Or is it specifically the SQL image that is not compatible?

Testcontainers for .NET supports both. I ran it primarily with WCOW to test legacy Windows applications and MSIs. This particular case is all about the Docker image. Multi arch images that follow the [Windows container version compatibility](https://learn.microsoft.com/en-us/virtualization/windowscontainers/deploy-containers/version-compatibility) run fine.

> The original integration tests only ran on linux anyways. So maybe what we could do is introduce a new xUnit attribute that will skip the test if not linux.

👍